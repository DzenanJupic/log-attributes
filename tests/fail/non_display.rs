#[log_attributes::log(info, "{fn} was executed: {return}")]
fn non_display() {}

fn main() {}
