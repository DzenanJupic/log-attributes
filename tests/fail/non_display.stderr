error[E0277]: `()` doesn't implement `std::fmt::Display`
 --> tests/fail/non_display.rs:2:18
  |
1 | #[log_attributes::log(info, "{fn} was executed: {return}")]
  | ----------------------------------------------------------- in this procedural macro expansion
2 | fn non_display() {}
  |                  ^^ `()` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `()`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: this error originates in the macro `format_args` (in Nightly builds, run with -Z macro-backtrace for more info)
